src/metabase/models/field_values.clj

```diff
@@ -321,35 +341,72 @@
 ;;; |                                                    CRUD fns                                                    |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defn- distinct-text-field-rff
```

Comment:
it might be nice to add a schema to this, to make the coupling to the QP explicit. then if the QP's `rff` type changes we'll get a schema error:
```suggestion
(mu/defn- distinct-text-field-rff :- ::qp.schema/rff
```

Reply:
added the schema here 4e6f6e6de4